client/src/components/Donation/stripe-card-form.tsx

```diff
@@ -0,0 +1,159 @@
+  const [paymentInfoValidation, setPaymentValidity] =
+    useState<PaymentInfoValidation>([
+      {
+        elementType: 'cardNumber',
+        complete: false,
+        error: null
+      },
+      {
+        elementType: 'cardExpiry',
+        complete: false,
+        error: null
+      }
+    ]);
+  const isPaymentInfoValid = paymentInfoValidation.every(
+    ({ complete, error }) => complete && !error
+  );
+
+  const stripe = useStripe();
+  const elements = useElements();
+
+  function handleInputChange(
+    event:
+      | StripeCardNumberElementChangeEvent
+      | StripeCardExpiryElementChangeEvent
+  ) {
+    const { elementType, error, complete } = event;
+    setPaymentValidity(
+      paymentInfoValidation.map(element => {
+        if (element.elementType === elementType)
+          return { elementType, error, complete };
+        return element;
+      })
+    );
+  }
+
+  const options = {
+    style: {
+      base: {
+        fontSize: '18px',
+        color: `${theme === 'night' ? '#fff' : '#0a0a23'}`
+      }
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!isPaymentInfoValid) return setSubmissionValidity(false);
+    else setSubmissionValidity(true);
+
+    if (!isSubmitting && stripe && elements) {
+      const cardElement = elements.getElement(CardNumberElement);
+      if (cardElement) {
+        const { error, token } = await stripe.createToken(cardElement);
+        if (error) {
+          onDonationStateChange({
+            redirecting: false,
+            processing: false,
+            success: false,
+            error: t('donate.went-wrong')
+          });
+        } else if (token) postStripeCardDonation(token);
+      }
+    }
+    return setSubmitting(false);
+  };
+
+  return (
+    <Form className='donation-form' onSubmit={handleSubmit}>
+      <div
+        className={`donation-elements${
+          !isSubmissionValid ? ' failed-submition' : ''
+        }`}
+      >
+        <CardNumberElement
+          className='form-control donate-input-element'
+          onChange={handleInputChange}
+          options={options}
+        />
+        <CardExpiryElement
+          className='form-control donate-input-element'
+          onChange={handleInputChange}
+          options={options}
+        />
+      </div>
+      <div className={'form-status'}>
+        {!isSubmissionValid && <p>{t('donate.valid-card')}</p>}
+      </div>
+      <Button
+        block={true}
+        bsStyle='primary'
+        className='confirm-donation-btn'
+        disabled={!stripe || !elements || isSubmitting}
+        type='submit'
+      >
+        Donate
+      </Button>
+    </Form>
+  );
+};
+
+const CardFormWrapper = (props: FormPropTypes): JSX.Element | null => {
+  if (!stripePublicKey) {
+    return null;
+  } else {
+    const stripePromise = loadStripe(stripePublicKey);
+    return (
+      <Elements stripe={stripePromise}>
```

Comment:
```suggestion
    return (
      <Elements stripe={loadStripe(stripePublicKey)}>
```
Since it's only used once.